packages/terra-date-input/src/DateInput.jsx

```diff
@@ -395,6 +401,14 @@ class DateInput extends React.Component {
     const inputValue = event.target.value;
     const stateValue = this.state.month;
     const maxValue = 12;
+    const currentMonth = moment().format('MM');
+
+    if (inputValue === currentMonth) {
+      const date = moment().format('YYYY-MM-DD');
```

Comment:
Does this format('YYYY-MM-DD') holds good for all locale

Reply:
Yes its working as expected for all locales

Reply:
Because of this hardcoded format value you onchange will always have date formatted with YYYY-MM-DD even if the supported format for locale is DD-MM-YYYY as shown in below screenshots

<img width="319" alt="Screenshot 2024-01-03 at 12 49 56 AM" src="https://github.com/cerner/terra-framework/assets/12869809/c0706879-3d97-4c2e-aaba-8a07dd4b96e0">
 
Format as per locale is MM-DD-YYYY but because of hard coded format date is returned with YYYY-MM-DD format:
<img width="233" alt="Screenshot 2024-01-03 at 12 52 33 AM" src="https://github.com/cerner/terra-framework/assets/12869809/ffd929a4-e5ec-4edd-ac58-069beefcfabd">